tutorials-v5/time-evolution/023_v5_paper-mesolve.md

```diff
@@ -0,0 +1,575 @@
+                   sigmam, sigmax, sigmaz, spost, spre, sprepost)
+
+# from qutip.solver.heom import HEOMSolver, BosonicBath
+
+%matplotlib inline
+```
+
+## Part 0: Introduction to the New Solver Class
+
+In our first example, we want to look at two interacting qubits that are (for now) decoupled from an environment.
+Such a system is described by the Hamiltonian
+
+$H = \dfrac{\epsilon_1}{2} \sigma_z^{(1)} + \dfrac{\epsilon_2}{2} \sigma_z^{(2)} + g \sigma_{x}^{(1)} \sigma_{x}^{(2)}$.
+
+The Pauli matrices $\sigma_z^{(1/2)}$ describe the respective two-level system of the qubits.
+The qubits are coupled via $\sigma_{x}^{(1/2)}$ and their interaction strength is given by $g$.
+
+```python
+epsilon1 = 1.0
+epsilon2 = 1.0
+g = 0.1
+
+sx1 = sigmax() & qeye(2)
+sx2 = qeye(2) & sigmax()
+sz1 = sigmaz() & qeye(2)
+sz2 = qeye(2) & sigmaz()
+
+H = (epsilon1 / 2) * sz1 + (epsilon2 / 2) * sz2 + g * sx1 * sx2
+
+print(H)
+```
+
+The dynamics of such a system is described by the Schrödinger equation
+
+$i \hbar \dfrac{d}{dt} \ket{\psi} = H \ket{\psi}$.
+Therefore, we can use `SESovler` to calculate the dynamics.
+
+```python
+se_solver = SESolver(H)
+psi0 = basis(2, 0) & basis(2, 1)
+tlist = np.linspace(0, 40, 100)
+```
+
+```python
+se_res = se_solver.run(psi0, tlist, e_ops=[sz1, sz2])
+```
+
+```python
+plt.plot(tlist, se_res.expect[0], label="i=1")
+plt.plot(tlist, se_res.expect[1], label="i=2")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma_z^{(i)} \rangle$")
+plt.legend()
+plt.show()
+```
+
+### Manual Stepping Interface
+
+A new feature in QuTiP v5 is that time steps can be controlled manually.
+This is specficially useful if the Hamiltonian depends on external control parameters such as field strength.
+Such parameters can be updated in euch step using the optional paramter `args`.
+In paractice, this can look like this:
+
+```python
+t = 0
+dt = 40 / 100
+se_solver2 = SESolver(H)
+se_solver2.start(psi0, t)
+while t < 40:
+    t = t + dt
+    psi = se_solver2.step(t)
+    # process the result psi and calculate next time step
+```
+
+### Solver and Integrator Options
+
+Another change in QuTiP v5 is that the `options` argument takes a standard Python dictionary.
+This should increase future felxibility and allow different solvers to provide individual sets of options more easily.
+The complete list of options can be found in the online documentation for each solver.
+
+As an example of frequently used options, we show `store_states`, determining whether the output should include the system state at each time step and `store_final_state` determening the same but for the final state.
+`method` is another common option, specifying the ODE integration method as well as its specific options.
+Also shown here are `atol` to control the precision (absolute tolerance),
+`nsteps` controls the maximum number of steps between two time steps and `max_step` giving the maximum allowed integration step of the default Adams ODE.
```

Comment:
Maybe good to add a sentence here like "Changing `max_step` can be particularly important for driven problems, to avoid the ODE solver over-stepping small sudden changes in the drive function."